src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -471,3 +471,55 @@ lemma ideal.homogeneous_hull_eq_Inf (I : ideal A) :
 eq.symm $ is_glb.Inf_eq $ (ideal.homogeneous_hull.gc 𝒜).is_least_l.is_glb
 
 end galois_connection
+
+section irrelavent_ideal
+
+open graded_algebra finset set_like.graded_monoid
+
+variables [comm_semiring R] [semiring A]
+variables [algebra R A] [decidable_eq ι] [add_monoid ι]
+variables (𝒜 : ℕ → submodule R A) [graded_algebra 𝒜]
+
+/--
+For an `ℕ`-graded ring `⨁ᵢ 𝒜ᵢ`, the irrelavent ideal refers to `⨁_{i≥0} 𝒜ᵢ`.
+-/
+def homogeneous_ideal.irrelavent : homogeneous_ideal 𝒜 :=
+let I : ideal A :=
+  { carrier := {a | proj 𝒜 0 a = 0 },
```

Comment:
Can this be expressed with `ring_hom.ker`? Is `proj 𝒜 0` upgradeable a ring homomorphism?

I'm a little surprised this is an ideal at all; `clifford_algebra.ι Q v` is in this carrier as far as I can tell, but `clifford_algebra.ι Q v * clifford_algebra.ι Q v` isn't. What am I missing?

Reply:
Yes, `proj 𝒜 0` is a ring homomorphism in the case of `ℕ`-grading.

I think definition of irrelevant ideal can be generalised to an arbitrary `linear_ordered_cancel_add_comm_monoid` by taking carrier to be `{a | ∀ (i : ι), i ≤ 0 → proj 𝒜 i a = 0}`. But I don't know if this is worth it.